sky/adaptors/aws.py

```diff
@@ -27,7 +43,11 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+# Creating the session object is not thread-safe for boto3,
+# so we add a reentrant lock to synchronize the session creation.
+# Reference: https://github.com/boto/boto3/issues/1592
 @functools.lru_cache()
+@_synchronized
```

Comment:
nit: directly call `with _session_creation_lock:` instead of decorator (to avoid other funcs using the same decorator erroneously)?

Reply:
Good point! Fixed. Thanks!